src/main/java/com/puppycrawl/tools/checkstyle/xpath/XpathQueryGenerator.java

```diff
@@ -332,4 +332,61 @@ private boolean isMatchingByLineAndColumnAndTokenType(DetailAST ast) {
                 && expandedTabColumn(ast) == columnNumber
                 && (tokenType == 0 || tokenType == ast.getType());
     }
+
+    /**
+     * Controls true escape characters. True escape characters need '&amp;' before but it also
+     * requires XML 1.1 until https://github.com/checkstyle/checkstyle/issues/5168.
+     *
+     * @param chr character to check.
+     * @param encStr StringBuilder from parent method encode().
+     */
+    private static void controlTrueCharacter(char chr, StringBuilder encStr) {
+        if (Character.isISOControl(chr)) {
+            encStr.append("#x");
+            encStr.append(Integer.toHexString(chr));
+            encStr.append(';');
+        }
+        else {
+            encStr.append(chr);
+        }
+    }
+
+    /**
+     * Escape &lt;, &gt;, &amp;, &#39; and &quot; as their entities.
+     *
+     * @param value the value to escape.
+     * @return the escaped value if necessary.
+     */
+    public static String encode(String value) {
+        final StringBuilder sb = new StringBuilder(256);
+        for (int i = 0; i < value.length(); i++) {
+            final char chr = value.charAt(i);
```

Comment:
we do not use "i"
can we use `value.chars().forEach` ?

Reply:
I used the regular for each for string, I hope that might be ok.

Reply:
Sir, this is also resolved according to me, as I have replaced the normal for against a foreach loop.

Reply:
Your approach is using copy of memory, 
Steams approach does not make copy 
https://github.com/bpupadhyaya/openjdk-8/blob/45af329463a45955ea2759b89cb0ebfe40570c3f/jdk/src/share/classes/java/lang/CharSequence.java#L130

Reply:
not done, but https://github.com/checkstyle/checkstyle/pull/9247#discussion_r576457648 looks like a way out.

Reply:
I had used value.chars().forEach, but there is a check which prevents for than 10 lines for `Lambda Expression` so I had to make a new method and then use it. Please tell if it works or nor?